processing/src/main/java/io/druid/segment/incremental/IncrementalIndexStorageAdapter.java

```diff
@@ -362,6 +387,10 @@ public void reset()
                 done = !foundMatched && (emptyRange || !baseIter.hasNext());
               }
 
+              private boolean beyondMaxRowIndex(Integer rowIndex) {
```

Comment:
rowIndex has to get unboxed anyway for comparison, but using `compareTo` additionally forces boxing of maxRowIndex. So accepting a primitive argument at least avoids the boxing of maxRowIndex.